### PR TITLE
Simplify the process of overriding spark-defaults

### PR DESCRIPTION
**Pull request by cjonesy**
- Environment variable overrides for spark-defaults.conf are more explicit
- Fixes problems with quoted arguments in entrypoint.sh
- Updated README.md and CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
```
Comment (dantswain) [inline CONTRIBUTING.md]: FWIW as an open source maintainer, I find I get a fair number of PRs out of the blue that are actually pretty useful.  You could still call this out as an open option, though - "feel free to solicit feedback" or whatever.

This is 100% up to you, though.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+| spark.local.dir               | SPARK_LOCAL_DIR               | /tmp                     |
+| spark.logConf                 | SPARK_LOGCONF                 | true                     |
+| spark.master                  | SPARK_MASTER                  | local                    |
+| spark.driver.supervise        | SPARK_DRIVER_SUPERVISE        | false                    |
+| spark.driver.extraClassPath   | SPARK_DRIVER_EXTRACLASSPATH   | /jars/*                  |
+| spark.executor.extraClassPath | SPARK_EXECUTOR_EXTRACLASSPATH | /jars/*                  |
+| spark.python.worker.memory    | SPARK_PYTHON_WORKER_MEMORY    | 512m                     |
+| spark.ui.enabled              | SPARK_UI_ENABLED              | true                     |
+| spark.eventLog.enabled        | SPARK_EVENTLOG_ENABLED        | false                    |
+| spark.eventLog.dir            | SPARK_EVENTLOG_DIR            | file:///tmp/spark-events |
```
Comment (dantswain) [inline README.md:58]: This is really nice :+1:.  A lot of people don't provide this level of documentation.

**Review by dantswain — APPROVED** [review]
I _think_ this is fine.  I've run into issues when people do too much in entrypoint scripts.  For example what if I just want to mount a spark-default.conf file of my own?  I think in your case, though, it's ok because they can just not define the variables that they don't want replaced.

File: CONTRIBUTING.md (hunk 1 above)
Comment (cjonesy) [reply CONTRIBUTING.md]: Yeah I was one the fence there... I think I'll drop that line.  We can always discuss it in the PR.

**Comment by cjonesy** [comment]
@dantswain 
RE: The complexity of `entrypoint.sh`

Would it be better if did something like this?
- Copy `spark-defaults.conf` to `spark-defaults.conf.docker`
- In `entrypoint.sh`, detect if someone has mounted their own spark-defaults.conf
  - If one is mounted, just use it
  - If one is not mounted, copy `spark-defaults.conf.docker` to `spark-default.conf` and do overrides

**Comment by dantswain** [comment]
I don't think you can detect if a volume is mounted at a point.  If you did it would be kind of breaking the encapsulation of the container anyways, I think.

I think what you have is fine, it's just a thing to be weary of.

**Pull request merged**